crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs

```diff
@@ -355,13 +366,17 @@ struct BindImportsAndExportsContext<'a> {
   pub options: &'a SharedOptions,
   pub errors: Vec<BuildDiagnostic>,
   pub warnings: Vec<BuildDiagnostic>,
+  pub external_import_binding_merger:
+    FxHashMap<ModuleIdx, FxHashMap<CompactStr, IndexSet<SymbolRef>>>,
 }
 
 impl BindImportsAndExportsContext<'_> {
+  #[allow(clippy::too_many_lines)]
   fn match_imports_with_exports(&mut self, module_id: ModuleIdx) {
     let Module::Normal(module) = &self.index_modules[module_id] else {
       return;
     };
+    let is_esm = matches!(self.options.format, OutputFormat::Esm);
```

Comment:
Only merge `import` binding in esm format, for other format it is not necessary.

Reply:
because the namedImport is already merged for other format